Initial work to use the protectedplanet database

drewpotter opened the pull request:
This PR is the initial work to use the ProtectedPlanet database as we merge over to use it instead.

Levia commented on app/models/country.rb:
```diff
@@ -1,5 +1,16 @@
+  # has_one :pame_statistic
+
+  belongs_to :region
+  belongs_to :region_for_index, -> { select('regions.id, regions.name') }, :class_name => 'Region', :foreign_key => 'region_id'
```
Why are those `region` and `region_for_index` instead of `country`? I see that names as `country` within the PP repo. So I would replicate that as much as possible

Levia replied:
Ah sorry, my bad. This is within the country model, so that's correct :)

Levia commented on config/database.yml:
```diff
@@ -1,20 +1,25 @@
+  host: <%= ENV['POSTGRES_HOST'] %>
+  username: <%= ENV['POSTGRES_USER'] %>
+  password: <%= ENV['POSTGRES_PASSWORD'] %>
+  template: template_postgis
```
Not sure what this template is for.
Also, I wonder why you have got those as ENV variables.

Looking at the PP database config file, that is a bit simpler then this. I would keep that structure for consistency.

https://github.com/unepwcmc/ProtectedPlanet/blob/master/config/database.yml

drewpotter replied:
Sorry @Levia I accidentally committed my Docker specific database settings here.

drewpotter replied:
@Levia would it be acceptable to simply copy and paste the database yml file from PP as I have the problem that I cannot run PP natively so I need that Docker ENV var style configuration to get it working locally. At least I know that the PP database yml is working.

Levia replied:
@drewpotter Yeah I guess that's fine. The db configuration should be the same anyway

Levia commented on config/database.yml:
```diff
@@ -1,20 +1,25 @@
 
+development:
+  <<: *default
+  database: protectedplanet-db
```
Can you also keep the name consistend? I would say `pp_development`

drewpotter replied:
Sorry @Levia this again is my Docker specific settings which I committed by mistake.

Levia commented:
I think there are some more tiny bits to fix:

* The rake task for the import it's still referencing the old model names (e.g. Evaluation instead of PameEvaluation)

* Not sure why my `db` folder still got one migration pending about the indexes, instead of having the entire folder removed completely.

* I would give it a try to include the db submodule here as well until we merge it completely with PP

drewpotter commented:
@Levia thanks I need to delete the rake task completely as we now do the import inside ProtectedPlanet instead. This is why it references the old model names, as I haven't updated this.

As for the `db` folder I think I can possibly delete it completely or as you suggested to include the `db` submodule instead.

Levia commented:
@drewpotter As this is now pointing to the PP db, it's fine to retain the import task here as well until we merge this completely. Is the import task already within PP?

I would give it a go about the db submodule

drewpotter commented:
@Levia the import task is in PP because @spencerldixon has implemented already PameEvaluations in PP. I made an assumption here, possibly the wrong one... which is that I would extend Spencer's work to replace the CSV he has which is partially similar to the CSV I am importing with fewer columns, with the CSV with the same data with extra columns. I then added the additional fields to that importer Spencer had built so we could do the PP-PAME import within PP. I thought - I assumed that we would want to automate this as much as possible and do the updates in one place. However I might have been completely wrong in that, so apologies if this is the wrong place for the importer. However, we do have some duplicated data in that case. If we want to have the importer in two places using the same (very similar) data.

Levia commented:
@drewpotter It's absolutely fine to have this on PP as well now.
I was just saying we could leave it here as well if just amending the model name is enough.

Otherwise we can just remove it completely from here

drewpotter commented:
@Levia I have removed the rake task and fixed the database settings, however, when I try to remove the database db folder git is not picking up that deletion. So I am not sure about making the db submodule for now due to that.

Levia commented:
@drewpotter The db folder doesn't seem to be in the repo for this branch though. So it should be fine. Not sure why I still had something in there

drewpotter commented:
Many thanks @Levia please can you re-check this PR to see if it is all okay now?

Levia commented:
@drewpotter Can you please remove the commented relationships from the models?
Happy to merge this after that
